packages/middleware-flexible-checksums/src/validateChecksumFromResponse.ts

```diff
@@ -35,7 +35,7 @@ export const validateChecksumFromResponse = async (
       const { base64Encoder } = config;
 
       if (isStreaming(responseBody)) {
-        createChecksumStream({
+        response.body = createChecksumStream({
```

Comment:
now that the body stream has a checksum wrapper, the section below where we perform the checksum validation should be removed

```js
      const checksum = await getChecksum(responseBody, { checksumAlgorithmFn, base64Encoder });
      if (checksum === checksumFromResponse) {
        // The checksum for response payload is valid.
        break;
      }
      throw new Error(
        `Checksum mismatch: expected "${checksum}" but received "${checksumFromResponse}"` +
          ` in response header "${responseHeader}".`
      );
```

this function should be renamed `validateChecksumFromResponse` -> `applyChecksumStreamWrapper` or something like that to reflect its new functionality.

Reply:
the ChecksumStream wrapper will do the error-throwing when the user reads the stream to completion

Reply:
The wrapper is used only when response.body is streaming. That's why there's a return on line 45.
For non-streaming use cases, the stringHasher needs to validate the checksum.

Reply:
Oh, I understand now